mono/metadata/threads.c

```diff
@@ -1837,6 +1837,13 @@ ves_icall_System_Threading_Thread_Join_internal(MonoThread *this_obj, int ms)
 	mono_error_set_pending_exception (&error);
 
 	if(ret==MONO_THREAD_INFO_WAIT_RET_SUCCESS_0) {
+#ifdef HOST_WIN32
```

Comment:
@vargaz IIUC `mono_join_uninterrupted` waits for an event to be signalled in the other thread (in `mono_threads_signal_thread_handle` which is called by `unregister_thread`). When the event is signalled the native thread has not yet been fully terminated.

@kumpera I guess you mean doing something similar to https://github.com/mono/mono/blob/master/mono/metadata/threads.c#L5048? Is that entire sequence starting with `MONO_ENTER_GC_SAFE` required?